src/components/ModalPortal.js

```diff
@@ -102,31 +112,32 @@ export default {
 
       'click-backdrop': () => this.$emit('click-backdrop')
     }
-
-    if (modal) {
-      return createModalVNode(
-        h,
-        {
-          props: modal.props,
+    const modals = this.getModals()
+      .map(({ name }) => this.modals[name])
+      .filter(v => v)
+
+    // for transition
+    const emptyModal = { children: [] }
+
+    return createModalVNode(
+      h,
+      [...modals, emptyModal].map(({ props, children, backdrop }) => ({
+        data: {
+          props,
           on: events
         },
-        modal.children,
-        modal.backdrop
-      )
-    } else {
-      return createModalVNode(h, { on: events }, [])
-    }
+        children,
+        backdrop
+      }))
+    )
   }
 }
 
-function createModalVNode(
-  h: Function,
-  data: any,
-  children: any[],
-  backdrop: ?any
-) {
+function createModalVNode(h: Function, modals: ModalData[]) {
   return h('div', { staticClass: 'modal-wrapper' }, [
-    h(Backdrop, data, backdrop),
-    h(ModalContent, data, children)
+    ...modals.map(({ data, children, backdrop }) => [
```

Comment:
Let's just use the return value of `map` method. We don't need to combine it to another array instance.